### PR TITLE
AMDGPU: Remove flat/global atomic fadd v2bf16 intrinsics

### PR DESCRIPTION
**Pull request by arsenm**
These are now fully covered by atomicrmw.

### CODE REVIEW

**Comment by arsenm** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/97050?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#97050** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/97050?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#96876** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96876?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#96875** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96875?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#96874** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96874?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#96873** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96873?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#96872** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96872?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#96760** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96760?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#97048](https://github.com/llvm/llvm-project/pull/97048) <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/97048?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#96759** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96759?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#96444** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96444?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#96443** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96443?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#96442** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96442?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#95930** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/95930?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#95929** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/95929?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @arsenm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by arsenm** [comment]
ping

File: llvm/test/CodeGen/AMDGPU/GlobalISel/fp-atomics-gfx940.ll [hunk 1]
```diff
@@ -106,106 +100,6 @@ define <2 x half> @flat_atomic_fadd_v2f16_rtn(ptr %ptr, <2 x half> %data) {
...
-; GFX940-LABEL: flat_atomic_fadd_v2bf16_noret:
-; GFX940:       ; %bb.0:
-; GFX940-NEXT:    s_load_dwordx2 s[0:1], s[2:3], 0x24
-; GFX940-NEXT:    s_load_dword s4, s[2:3], 0x2c
-; GFX940-NEXT:    s_waitcnt lgkmcnt(0)
-; GFX940-NEXT:    v_mov_b64_e32 v[0:1], s[0:1]
-; GFX940-NEXT:    v_mov_b32_e32 v2, s4
-; GFX940-NEXT:    flat_atomic_pk_add_bf16 v[0:1], v2
-; GFX940-NEXT:    s_endpgm
-  %ret = call <2 x i16> @llvm.amdgcn.flat.atomic.fadd.v2bf16.p0(ptr %ptr, <2 x i16> %data)
```
Comment (yxsamliu) [inline llvm/test/CodeGen/AMDGPU/GlobalISel/fp-atomics-gfx940.ll:119]: do we have equivalent codegen tests for the counterpart atomicrmw insts to cover the removed tests? same as below

Comment (arsenm) [reply llvm/test/CodeGen/AMDGPU/GlobalISel/fp-atomics-gfx940.ll:119]: Yes

**Comment by arsenm** [comment]
ping

**Review by yxsamliu — APPROVED** [review]

**Comment by arsenm** [comment]
### Merge activity

* **Aug 21, 6:23 AM EDT**: @arsenm started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/llvm/llvm-project/97050).

**Pull request merged**